src/gridfs/upload.ts

```diff
@@ -194,7 +216,10 @@ export class GridFSBucketWriteStream extends Writable {
     }
 
     this.state.aborted = true;
-    await this.chunks.deleteMany({ files_id: this.id });
+    const remainingTimeMS = this.timeoutContext?.getRemainingTimeMSOrThrow(
+      `Upload timed out after ${this.timeoutContext?.timeoutMS}ms`
+    );
+    await this.chunks.deleteMany({ files_id: this.id, timeoutMS: remainingTimeMS });
   }
```

Comment:
Is this really the correct behavior?  This would leave the user's data in a partially-written state.  That doesn't seem right

Reply:
This is the behaviour expected by the spec([1](https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.md#gridfs-api), [2](https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.md#gridfs-streams-behavior), [3](https://github.com/mongodb/specifications/tree/master/source/client-side-operations-timeout/tests#aborting-an-upload-stream-can-be-timed-out)), but I do see your point here as it could leave user's collections in an unexpected state. I can see an argument for treating `abort` the same way we treat `close` for cursors in that it seems more user friendly to refresh the timeout before calling `deleteMany`

Reply:
Filed DRIVERS-2996